### DISCUSSION TITLE
Literal text inside role content

### DISCUSSION THREAD

**Discussion by astrojuanlu** [executablebooks/meta#1141] [Q&A] [answered]
Is it possible to do something like

```
{py:func}`the Python \`dir\` function <dir>`
```

?

**Comment by agoose77** [comment]
@astrojuanlu in theory, Sphinx allows this, and this Python domain role is defined by Sphinx. As such, we should support it too. However, there's always the possibility that it's not been tested yet. I assume you're talking about Jupyter Book and/or the Sphinx Executable Books tools here? 

Have you tried it? :)

**Reply by astrojuanlu** [reply]
Thanks for the prompt reply! The reason I asked is that, when I went to try it, my IDE (VSCode) got really confused:

![image](https://github.com/executablebooks/meta/assets/316517/0e49588e-c0ac-4103-8f93-b41bcef9423b)

And indeed, the output gets messed up:

![image](https://github.com/executablebooks/meta/assets/316517/6edf6214-3d91-4b1c-9531-ffbd9bc87d5a)

**Reply by astrojuanlu** [reply]
FTR I'm using `Sphinx==6.2.1` with `myst-parser==1.0.0`

**Reply by agoose77** [reply, marked as answer]
Ah, I can see there's a nuance that I missed - the _literal_ `dir`. I believe this is [a limitation or ReST / inline roles](https://documatt.com/restructuredtext-reference/element/inline-literal.html#no-inline-element-nesting). Whilst it looks like MyST parses this "wrong", the result would still not work according to this rule.

Pinging @chrisjsewell for his thoughts on the general "should we support this" question.

**Reply by rowanc1** [reply]
Does it work if you surround with two `` ` `` ?
```
{py:func}``the Python `dir` function <dir>``
```

(that might just get it past the myst parsing, not the sphinx side though!)

**Reply by astrojuanlu** [reply]
Fixed the parsing, didn't get me what I wanted 😅 

![image](https://github.com/executablebooks/meta/assets/316517/c142ccac-1391-4dcc-83c8-81f40a12e97d)
